docs/resources/host_catalog_plugin.md

```diff
@@ -94,6 +94,7 @@ resource "boundary_host_catalog_plugin" "azure_example" {
 - `plugin_name` (String) The name of the plugin that should back the resource. This or plugin_id must be defined.
 - `secrets_hmac` (String) The HMAC'd secrets value returned from the server.
 - `secrets_json` (String, Sensitive) The secrets for the host catalog. Either values encoded with the "jsonencode" function, pre-escaped JSON string, or a file:// or env:// path. Set to a string "null" to clear any existing values. NOTE: Unlike "attributes_json", removing this block will NOT clear secrets from the host catalog; this allows injecting secrets for one call, then removing them for storage.
+- `worker_filter` (String) HCP Only. A filter used to control which PKI workers can handle dynamic host catalog requests.
```

Comment:
Interesting I wonder if we need to update our other usage of HCP Only - this is a relic of before we had self hosted. Perhaps we should align this with Enterprise-only as we did in Boundary. But perhpas worth starting a thread and taggin Anando from product on his thoughts

Reply:
Yeah, I really only followed our current convention in these docs, but this is technically incorrect.

I suppose we could update it everywhere to say `HCP and Boundary Enterprise only` or something like this. @anando-chatterjee WDYT?

Reply:
(Also added a CHANGELOG entry with https://github.com/hashicorp/terraform-provider-boundary/compare/9d5d0ec74ba5c995e2674612074382ac974d7f3f..fc7f471ebf2aaddfc965552813b0ab6bdd713f9d)

Reply:
Updated to Boundary v0.18.0
https://github.com/hashicorp/terraform-provider-boundary/compare/fc7f471ebf2aaddfc965552813b0ab6bdd713f9d..4efc3734c0003906ab001cfdadb4e8924355ff51